rds-update-rdsh-collection/azuredeploy.json

```diff
@@ -0,0 +1,318 @@
+
+        "impersonateScript": "https://gallery.technet.microsoft.com/scriptcenter/Impersonate-a-User-9bfeff82/file/127189/1/New-ImpersonateUser.ps1",
+
+        "p": {
+            "domain": "[concat(' -domain ', parameters('existingDomainName'))]",
+            "user": "[concat(' -username ', parameters('existingAdminUsername'))]",
+            "pwd": "[concat(' -password ', parameters('existingAdminPassword'))]",
+            "collection": "[concat(' -collection \"', parameters('existingRdshCollectionName'), '\"')]",
+            "iteration": "[concat(' -iteration ', toLower(parameters('rdshUpdateIteration')))]",
+            "servers": "[concat(' -nServers ', parameters('rdshNumberOfInstances'))]",
+            "timeout": "[concat(' -nTimeoutMinutes ', parameters('userLogoffTimeoutInMinutes'))]"
+        },
+        "scriptParameters": "[concat(variables('p').domain, variables('p').user, variables('p').pwd, variables('p').collection, variables('p').iteration, variables('p').servers, variables('p').timeout)]",
+        "scriptFolder": "Scripts",
+        "scriptFileName":  "Script.ps1"
+    },
+
+    "resources": [
+
+        {
+            "apiVersion": "2015-06-15",
+            "type": "Microsoft.Network/networkInterfaces",
+            "name": "[concat('rdsh-', copyindex(),variables('iteration'), '-nic')]",
+            "location": "[resourceGroup().location]",
+            "copy": {
+                "name": "rdsh-nic-loop",
+                "count": "[parameters('rdshNumberOfInstances')]"
+            },
+            "properties": {
+                "ipConfigurations": [
+                    {
+                        "name": "ipconfig",
+                        "properties": {
+                            "privateIPAllocationMethod": "Dynamic",
+                            "subnet": {
+                                "id": "[concat(resourceId('Microsoft.Network/virtualNetworks', parameters('existingVnetName')), '/subnets/', parameters('existingSubnetName'))]"
+                            }
+                        }
+                    }
+                ]
+            }
+        },
+
+        {
+            "apiVersion": "2015-06-15",
+            "type": "Microsoft.Compute/virtualMachines",
+            "name": "[concat('rdsh-', copyindex(),variables('iteration'))]",
+            "location": "[resourceGroup().location]",
+            "copy": {
+                "name": "rdsh-vm-loop",
+                "count": "[parameters('rdshNumberOfInstances')]"
+            },
+            "dependsOn": [
+                "[concat('Microsoft.Network/networkInterfaces/', 'rdsh-', copyindex(),variables('iteration'), '-nic')]"
+            ],
+            "properties": {
+                "hardwareProfile": {
+                    "vmSize": "[parameters('rdshVmSize')]"
+                },
+                "availabilitySet": {
+                    "id": "[resourceId('Microsoft.Compute/availabilitySets', 'rdsh-availabilityset')]"
```

Comment:
is this a prereq?  it's not defined in the template and has a hard coded name...